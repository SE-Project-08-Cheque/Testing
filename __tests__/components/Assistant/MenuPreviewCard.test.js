// import dependencies
import React from "react";

// import react-testing methods
import { render, cleanup } from "@testing-library/react";

// add custom jest matchers from jest-dom
import "@testing-library/jest-dom";

// import react dom
import ReactDOM from "react-dom";
import { MemoryRouter } from "react-router-dom";


// import renderer for take snapshots
import renderer from "react-test-renderer";

// import the component for testing
import MenuPreviewCard from "../../../components/Assistant/MenuPreviewCard";

afterEach(cleanup);
it("renders without crashing", () => {
  const div = document.createElement("div");
    ReactDOM.render(
        <MemoryRouter>
            <MenuPreviewCard />
        </MemoryRouter>
        , div);
});

it("renders menu preview card correctly", () => {
    const { getByTestId } = render(
        <MemoryRouter>
            <MenuPreviewCard
                imageUrl="https://blogmedia.evbstatic.com/wp-content/uploads/wpmulti/sites/8/2017/04/21143242/tourists.jpg"
                imageAlt="Tourist Couple Picture"
                title="Tourist Manager"
                linkTo="/app/tourists"
                description="Add Tousrists, Delete Tourists, Update Tourists Information and View Activites"
            />
        </MemoryRouter>
    );
    expect(getByTestId('title')).toHaveTextContent("Tourist Manager");

});

it("matches snapshot", () => {
    const tree = renderer.create(
        <MemoryRouter>
        <MenuPreviewCard
                imageUrl="https://blogmedia.evbstatic.com/wp-content/uploads/wpmulti/sites/8/2017/04/21143242/tourists.jpg"
                imageAlt="Tourist Couple Picture"
                title="Tourist Manager"
                linkTo="/app/tourists"
                description="Add Tousrists, Delete Tourists, Update Tourists Information and View Activites"
        
  /></MemoryRouter>);

    expect(tree).toMatchSnapshot();
})

