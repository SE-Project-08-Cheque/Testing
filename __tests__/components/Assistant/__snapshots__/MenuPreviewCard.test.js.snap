// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<a
  as={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "propTypes": Object {
        "to": [Function],
      },
      "render": [Function],
    }
  }
  href="/app/tourists"
  onClick={[Function]}
>
  <div
    className="css-17xbt85"
    data-testid="MenuPreviewCard"
  >
    <img
      alt="Tourist Couple Picture"
      className="chakra-image__placeholder css-1rdn362"
    />
    <div
      className="css-mfzh6r"
    >
      <div
        className="css-0"
      >
        <div
          className="css-76gc8x"
          data-testid="title"
        >
          Tourist Manager
        </div>
        <div
          className="chakra-collapse"
          style={
            Object {
              "display": "none",
              "height": "0px",
              "opacity": 0,
              "overflow": "hidden",
            }
          }
        >
          <div
            className="css-j5j99d"
          >
            Add Tousrists, Delete Tourists, Update Tourists Information and View Activites
          </div>
        </div>
        <div
          className="css-k008qs"
        >
          <div
            className="css-0"
          />
          <div
            className="css-17xejub"
          />
          <div
            className="css-0"
          >
            <button
              className="chakra-button css-143x6ju"
              onClick={[Function]}
              type="button"
            >
              Show 
              More
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</a>
`;
